File: contracts/scripts/generate_keypair.js
#!/usr/bin/env node

// Script para generar un nuevo keypair de Stellar/Soroban
// Requiere: npm install @stellar/stellar-sdk

const { Keypair } = require('@stellar/stellar-sdk');

function generateKeypair() {
    // Generar un nuevo keypair aleatorio
    const keypair = Keypair.random();
    
    console.log('=== NUEVO KEYPAIR GENERADO ===');
    console.log('Public Key (Address):', keypair.publicKey());
    console.log('Secret Key (Hex):', keypair.rawSecretKey().toString('hex'));
    console.log('Secret Key (Stellar):', keypair.secret());
    console.log('');
    console.log('IMPORTANTE:');
    console.log('- Usa el "Secret Key (Hex)" para ADMIN_SECRET_KEY en tu .env');
    console.log('- Usa el "Public Key (Address)" para inicializar el contrato');
    console.log('- NUNCA compartas el secret key - guárdalo de forma segura');
    console.log('');
    console.log('Para usar en testnet, financia la cuenta en:');
    console.log('https://laboratory.stellar.org/#account-creator?network=test');
}

generateKeypair();

File: contracts/scripts/generate_keypair.py
#!/usr/bin/env python3

# Script para generar un nuevo keypair de Stellar/Soroban
# Requiere: pip install stellar-sdk

from stellar_sdk import Keypair

def generate_keypair():
    # Generar un nuevo keypair aleatorio
    keypair = Keypair.random()
    
    print('=== NUEVO KEYPAIR GENERADO ===')
    print(f'Public Key (Address): {keypair.public_key}')
    print(f'Secret Key (Hex): {keypair.raw_secret_key().hex()}')
    print(f'Secret Key (Stellar): {keypair.secret}')
    print('')
    print('IMPORTANTE:')
    print('- Usa el "Secret Key (Hex)" para ADMIN_SECRET_KEY en tu .env')
    print('- Usa el "Public Key (Address)" para inicializar el contrato')
    print('- NUNCA compartas el secret key - guárdalo de forma segura')
    print('')
    print('Para usar en testnet, financia la cuenta en:')
    print('https://laboratory.stellar.org/#account-creator?network=test')

if __name__ == '__main__':
    generate_keypair()